### ISSUE TITLE
Not working on 2021.12

### ISSUE THREAD

**Issue by jomavic** [Limych/ha-jq300#75]
**Environment**

- Home Assistant Core release with the issue:
- This custom component release with the issue:
- Last working this custom component release (if known):
- Operating environment (Home Assistant/Supervisor/Docker/venv): raspberry pi docker

**Describe the bug**
???? Not starting  

**Configuration.yaml**
```yaml

Add your configs here if any.

```

**Steps to Reproduce**
 error loading

**Expected behavior**

**Debug log**

```text

Add your logs here.

Logger: homeassistant.setup
Source: setup.py:153
First occurred: 17:45:02 (1 occurrences)
Last logged: 17:45:02

Setup failed for jq300: Unable to import component: cannot import name 'HTTP_OK' from 'homeassistant.const' (/usr/src/homeassistant/homeassistant/const

**Additional context**
<!--
  Add any other context about the problem here.
-->

**Comment by Tonguc-Endem** [comment]
Same here

When I add the config and check it says:
Component error: jq300 - cannot import name 'HTTP_OK' from 'homeassistant.const' (/usr/src/homeassistant/homeassistant/const.py)

**Comment by elik745i** [comment]
Yea mine says the same, please fix it I find the integration quite useful!

**Comment by Kurtroosbeek2** [comment]
Here the same problem.

**Comment by Kurtroosbeek2** [comment]
Solution:
Edit the file:
/config/custom_components/jq300/api.py

Around line 20 add the from line:

```
import async_timeout
from http import HTTPStatus
import homeassistant.util.dt as dt_util
```

Around line 25 change the code to delete the HTTP status OK.
```
from homeassistant.const import (
    CONCENTRATION_MILLIGRAMS_PER_CUBIC_METER,
    CONCENTRATION_PARTS_PER_BILLION,
    CONCENTRATION_PARTS_PER_MILLION,
    CONF_UNIT_OF_MEASUREMENT,
)
```


Around Line 216 change the code to this:
```
        if (
            response.status != HTTPStatus.OK and response.status != HTTP_NO_CONTENT
        ):  # pragma: no cover
            _LOGGER.debug(MSG_GENERIC_FAIL)
            return None
```

 That solved it for me :)

**Comment by LandZen** [comment]
> Solution: Edit the file: /config/custom_components/jq300/api.py
> 
> Around line 20 add the from line:
> 
> ```
> import async_timeout
> from http import HTTPStatus
> import homeassistant.util.dt as dt_util
> ```
> 
> Around line 25 change the code to delete the HTTP status OK.
> 
> ```
> from homeassistant.const import (
>     CONCENTRATION_MILLIGRAMS_PER_CUBIC_METER,
>     CONCENTRATION_PARTS_PER_BILLION,
>     CONCENTRATION_PARTS_PER_MILLION,
>     CONF_UNIT_OF_MEASUREMENT,
> )
> ```
> 
> Around Line 216 change the code to this:
> 
> ```
>         if (
>             response.status != HTTPStatus.OK and response.status != HTTP_NO_CONTENT
>         ):  # pragma: no cover
>             _LOGGER.debug(MSG_GENERIC_FAIL)
>             return None
> ```
> 
> That solved it for me :)

Work perfectly!! Thank you!

**Comment by elik745i** [comment]
> Solution: Edit the file: /config/custom_components/jq300/api.py
> 
> Around line 20 add the from line:
> 
> ```
> import async_timeout
> from http import HTTPStatus
> import homeassistant.util.dt as dt_util
> ```
> 
> Around line 25 change the code to delete the HTTP status OK.
> 
> ```
> from homeassistant.const import (
>     CONCENTRATION_MILLIGRAMS_PER_CUBIC_METER,
>     CONCENTRATION_PARTS_PER_BILLION,
>     CONCENTRATION_PARTS_PER_MILLION,
>     CONF_UNIT_OF_MEASUREMENT,
> )
> ```
> 
> Around Line 216 change the code to this:
> 
> ```
>         if (
>             response.status != HTTPStatus.OK and response.status != HTTP_NO_CONTENT
>         ):  # pragma: no cover
>             _LOGGER.debug(MSG_GENERIC_FAIL)
>             return None
> ```
> 
> That solved it for me :)

Excellent, thanks!

**Comment by panenka** [comment]
> Solution:
> Edit the file:

Many thanks, solved it for me too :)

**Comment by binoke** [comment]
Thx for this fix.  It worked for me as well.

**Comment by smartnoob** [comment]
Thank you. This worked for me as well.

**Comment by robikop** [comment]
not sure why but I don't see my entities anymore :(

I've added via hasc the integration changed the api.py based on the above and I don't have the error anymore but also all entities are gone.
What do I do wring?

This is my confimg:
jq300:
  username: ------
  password: ------
  devices: jq300
      
iaquk:
  jq300:
    sources:
      humidity: sensor.jq300_humidity
      co2: sensor.jq300_eco2
      tvoc: sensor.jq300_tvoc
      hcho: sensor.jq300_hcho
      pm: sensor.jq300_pm25


logs:
2021-12-28 21:48:05 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration jq300 which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant


2021-12-28 21:49:26 WARNING (MainThread) [custom_components.iaquk] Entity sensor.jq300_humidity not found
2021-12-28 21:49:26 WARNING (MainThread) [custom_components.iaquk] Entity sensor.jq300_eco2 not found
2021-12-28 21:49:26 WARNING (MainThread) [custom_components.iaquk] Entity sensor.jq300_tvoc not found
2021-12-28 21:49:26 WARNING (MainThread) [custom_components.iaquk] Entity sensor.jq300_hcho not found
2021-12-28 21:49:26 WARNING (MainThread) [custom_components.iaquk] Entity sensor.jq300_pm25 not found


What do I miss?
